components/core/spinning-text.tsx

```diff
@@ -0,0 +1,58 @@
+'use client';
+import { motion } from 'framer-motion';
+import React, { CSSProperties, ReactElement } from 'react';
+
+// Define the props types for the SpinningText component
+interface SpinningTextProps {
+  children: ReactElement;
+  width?: number;
+  textStyle?: CSSProperties;
+  degree?: number;
```

Comment:
can we add props:
- className
- duration

Reply:
I added both duration and className.